Sources/Sentry/Profiling/SentryProfilerSerialization.mm

```diff
@@ -196,9 +207,138 @@
     return payload;
 }
 
+NSMutableDictionary<NSString *, id> *
+sentry_serializedContinuousProfileChunk(
+    SentryId *profileID, NSDictionary<NSString *, id> *profileData, uint64_t startSystemTime,
+    uint64_t endSystemTime, NSDictionary<NSString *, id> *serializedMetrics,
+    NSArray<SentryDebugMeta *> *debugMeta, SentryHub *hub, uint64_t continuousChunkStartSystemTime
+#    if SENTRY_HAS_UIKIT
+    ,
+    SentryScreenFrames *gpuData
+#    endif // SENTRY_HAS_UIKIT
+)
+{
+    NSMutableArray<SentrySample *> *const samples = profileData[@"profile"][@"samples"];
+    // !!!: assumption: in legacy, we would avoid sending a payload with less than 2 samples. now,
+    // we may have previously sent a chunk with many samples, and then this chunk may be the last of
+    // the continuous profiling session and it only has 1 sample. assuming we'll want to keep that
+    // sample. let the backend decide, or, keep track of some statistics of the profiling session,
+    // to count the number of total samples sent so far, to decide whether or not a 1-sample chunk
+    // is acceptable.
+
+    const auto payload = [NSMutableDictionary<NSString *, id> dictionary];
+    NSMutableDictionary<NSString *, id> *const profile = [profileData[@"profile"] mutableCopy];
+    profile[@"samples"] = _sentry_serializedSamplesWithRelativeTimestamps(
+        samples, startSystemTime, SentryProfilerModeContinuous);
+
+    payload[@"profile"] = profile;
+
+    payload[@"version"] = @"2";
+    const auto debugImages = [NSMutableArray<NSDictionary<NSString *, id> *> new];
+    for (SentryDebugMeta *debugImage in debugMeta) {
+        [debugImages addObject:[debugImage serialize]];
+    }
+    if (debugImages.count > 0) {
+        payload[@"debug_meta"] = @ { @"images" : debugImages };
+    }
+
+    payload[@"chunk_id"] = [[[SentryId alloc] init] sentryIdString];
+    payload[@"profile_id"] = profileID.sentryIdString;
+    payload[@"environment"] = hub.scope.environmentString ?: hub.getClient.options.environment;
+    payload[@"release"] = hub.getClient.options.releaseName;
+    payload[@"platform"] = SentryPlatformName;
+
+    // add the gathered metrics
+    auto metrics = serializedMetrics;
+
+#    if SENTRY_HAS_UIKIT
+    const auto mutableMetrics =
+        [NSMutableDictionary<NSString *, id> dictionaryWithDictionary:metrics];
+    const auto slowFrames = sentry_sliceGPUData(gpuData.slowFrameTimestamps, startSystemTime,
+        endSystemTime, /*useMostRecentFrameRate */ NO, SentryProfilerModeContinuous);
+    if (slowFrames.count > 0) {
```

Comment:
`m`: Codecov complains quite a bit for uncovered lines in this method. Should we add tests for them?

Reply:
I'm beefing up the tests around this.

Reply:
I have another branch at the top of the stack where i was already adding more tests, I will do these there because these particular tests involve a lot of work around timestamps, and we just made a decision that's going to affect that (moving to using NSDate for all profiling timestamps), which I'd like to merge before writing the tests to avoid potentially doing a lot of annoying debugging twice.